shared/settings/chat/index.tsx

```diff
@@ -1,24 +1,47 @@
 import * as React from 'react'
 import * as RPCChatTypes from '../../constants/types/rpc-chat-gen'
 import * as Constants from '../../constants/settings'
+import {TeamDetails, TeamID} from '../../constants/types/teams'
 import * as Kb from '../../common-adapters'
 import * as Styles from '../../styles'
 
 export type Props = {
+  contactSettingsEnabled?: boolean
+  contactSettingsError: string
+  contactSettingsIndirectFollowees: boolean
+  contactSettingsSelectedTeams: {[K in TeamID]: boolean}
   unfurlMode?: RPCChatTypes.UnfurlMode
   unfurlWhitelist?: Array<string>
   unfurlError?: string
-  onUnfurlSave: (arg0: RPCChatTypes.UnfurlMode, array: Array<string>) => void
+  onContactSettingsSave: (
+    enabled: boolean,
+    indirectFollowees: boolean,
+    teamsEnabled: boolean,
+    teamsList: {[k in TeamID]: boolean}
+  ) => void
+  onUnfurlSave: (mode: RPCChatTypes.UnfurlMode, whitelist: Array<string>) => void
   onRefresh: () => void
+  teamDetails: Array<TeamDetails>
 }
 
 type State = {
+  contactSettingsEnabled?: boolean | null
+  contactSettingsIndirectFollowees?: boolean | null
+  contactSettingsSelectedTeams: {[K in TeamID]: boolean}
+  contactSettingsTeamsEnabled?: boolean | null
```

Comment:
Can't seem to understand when/why/where these would be null? state s initialized to `undefined` and I think the RPCs don't return `null`. Could be wrong here.

Reply:
@thebearjew Ah, I think it's the case where the load RPC failed and we set `contactSettingsError` but nothing else.